spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-binder-sample/src/main/java/com/example/SinkExample.java

```diff
@@ -35,7 +37,17 @@
 
 	@StreamListener(Sink.INPUT)
 	public void handleMessage(UserMessage userMessage) {
+		if (userMessage.isThrowError()) {
+			throw new RuntimeException("An error was triggered in the message handler!");
+		}
+
 		LOGGER.info("New message received from " + userMessage.getUsername() + ": " + userMessage.getBody() +
 				" at " + userMessage.getCreatedAt());
 	}
+
+	@StreamListener("errorChannel")
```

Comment:
I think its time to demonstrate a *destination-specific error handling* instead.
See that SCSt doc I've mentioned before: https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/2.2.1.RELEASE/spring-cloud-stream.html#spring-cloud-stream-overview-error-handling

The sample is:
```
@ServiceActivator(inputChannel = Processor.INPUT + ".myGroup.errors") //channel name 'input.myGroup.errors'
```

Reply:
Hey Artem, so to get this to work I ultimately had to do something like this:

```
@ServiceActivator(inputChannel = "inputTopic.myGroup.myGroup.errors")
```

But is there a way to not hardcode `inputTopic` and `myGroup`? We were hoping to let users configure the topic and group name through `application.properties`; however it appears I can only put constant values in the `ServiceActivator` for `inputChannel`.

Reply:
Not sure about `myGroup.myGroup`, but yes: this is only the way now. An error channel is named really after a destination mapping (see that `Processor.INPUT` above) an a group assigned into this binding target.